### ISSUE TITLE
GeoSeries.transform returns "unsupported operand type(s)"

### ISSUE THREAD

**Issue by floriandeboissieu** [geopandas/geopandas#3179]
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of geopandas.

- [ ] (optional) I have confirmed this bug exists on the main branch of geopandas.

---

#### Code Sample
Taken from the documentation: https://geopandas.org/en/latest/docs/reference/api/geopandas.GeoSeries.transform.html
```python
import geopandas
from shapely import Point, Polygon
s = geopandas.GeoSeries([Point(0, 0)])
s.transform(lambda x: x + 1)
# returns: TypeError: unsupported operand type(s) for +: 'GeometryArray' and 'int'
```

#### Problem description

Should run without error, or I did not understand well the documentation?

#### Expected Output
```python
0    POINT (1 1)
dtype: geometry
```

#### Output of ``geopandas.show_versions()``

<details>

SYSTEM INFO
-----------
python     : 3.12.1 | packaged by conda-forge | (main, Dec 23 2023, 08:03:24) [GCC 12.3.0]
executable : /home/jovyan/mambaforge/envs/geopandas/bin/python
machine    : Linux-5.14.0-1059-oem-x86_64-with-glibc2.31

GEOS, GDAL, PROJ INFO
---------------------
GEOS       : 3.12.1
GEOS lib   : None
GDAL       : 3.8.3
GDAL data dir: /home/jovyan/mambaforge/envs/geopandas/share/gdal
PROJ       : 9.3.1
PROJ data dir: /home/jovyan/mambaforge/envs/geopandas/share/proj

PYTHON DEPENDENCIES
-------------------
geopandas  : 0.14.3
numpy      : 1.26.4
pandas     : 2.2.0
pyproj     : 3.6.1
shapely    : 2.0.2
fiona      : 1.9.5
geoalchemy2: None
geopy      : None
matplotlib : 3.8.2
mapclassify: 2.6.1
pygeos     : None
pyogrio    : None
psycopg2   : None
pyarrow    : None
rtree      : 1.2.0

</details>

**Comment by martinfleis** [comment]
Can you show the full Traceback? I am unable to reproduce the issue in practically the same environment. I sense some installation issue is causing this.

**Comment by floriandeboissieu** [comment]
```python
Traceback (most recent call last):                                                                                                                                                   
  File "/home/jovyan/mambaforge/envs/geopandas/lib/python3.12/site-packages/pandas/core/apply.py", line 314, in transform_str_or_callable                                          
    return obj.apply(func, args=args, **kwargs)                                                                                                                                      
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                      
  File "/home/jovyan/mambaforge/envs/geopandas/lib/python3.12/site-packages/geopandas/geoseries.py", line 673, in apply                                                            
    result = super().apply(func, args=args, **kwargs)                                                                                                                                
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                
  File "/home/jovyan/mambaforge/envs/geopandas/lib/python3.12/site-packages/pandas/core/series.py", line 4904, in apply                                                            
    ).apply()                                                                                                                                                                        
      ^^^^^^^                                                                                                                                                                        
  File "/home/jovyan/mambaforge/envs/geopandas/lib/python3.12/site-packages/pandas/core/apply.py", line 1427, in apply
    return self.apply_standard()
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/jovyan/mambaforge/envs/geopandas/lib/python3.12/site-packages/pandas/core/apply.py", line 1507, in apply_standard
    mapped = obj._map_values(
             ^^^^^^^^^^^^^^^^
  File "/home/jovyan/mambaforge/envs/geopandas/lib/python3.12/site-packages/pandas/core/base.py", line 919, in _map_values
    return arr.map(mapper, na_action=na_action)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jovyan/mambaforge/envs/geopandas/lib/python3.12/site-packages/pandas/core/arrays/base.py", line 2319, in map
    return map_array(self, mapper, na_action=na_action)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jovyan/mambaforge/envs/geopandas/lib/python3.12/site-packages/pandas/core/algorithms.py", line 1743, in map_array
    return lib.map_infer(values, mapper, convert=convert)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "lib.pyx", line 2972, in pandas._libs.lib.map_infer
  File "<stdin>", line 1, in <lambda>
TypeError: unsupported operand type(s) for +: 'Point' and 'int'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jovyan/mambaforge/envs/geopandas/lib/python3.12/site-packages/pandas/core/series.py", line 4766, in transform
    result = SeriesApply(ser, func=func, args=args, kwargs=kwargs).transform()
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jovyan/mambaforge/envs/geopandas/lib/python3.12/site-packages/pandas/core/apply.py", line 246, in transform
    result = self.transform_str_or_callable(func)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jovyan/mambaforge/envs/geopandas/lib/python3.12/site-packages/pandas/core/apply.py", line 316, in transform_str_or_callable
    return func(obj, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<stdin>", line 1, in <lambda>
  File "/home/jovyan/mambaforge/envs/geopandas/lib/python3.12/site-packages/pandas/core/ops/common.py", line 76, in new_method
    return method(self, other)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/jovyan/mambaforge/envs/geopandas/lib/python3.12/site-packages/pandas/core/arraylike.py", line 186, in __add__
    return self._arith_method(other, operator.add)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jovyan/mambaforge/envs/geopandas/lib/python3.12/site-packages/pandas/core/series.py", line 6115, in _arith_method
    return base.IndexOpsMixin._arith_method(self, other, op)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jovyan/mambaforge/envs/geopandas/lib/python3.12/site-packages/pandas/core/base.py", line 1382, in _arith_method
    result = ops.arithmetic_op(lvalues, rvalues, op)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jovyan/mambaforge/envs/geopandas/lib/python3.12/site-packages/pandas/core/ops/array_ops.py", line 273, in arithmetic_op
    res_values = op(left, right)
                 ^^^^^^^^^^^^^^^
TypeError: unsupported operand type(s) for +: 'GeometryArray' and 'int'
```

**Comment by floriandeboissieu** [comment]
I have the same result installing geopandas in micromamba docker image:
```shell
docker run --rm -it mambaorg/micromamba bash
$ micromamba create -n geopandas -c conda-forge geopandas
$ python
>>> import geopandas
>>> from shapely import Point, Polygon
>>> s = geopandas.GeoSeries([Point(0, 0)])
>>> s.transform(lambda x: x + 1)
```


```

**Comment by martinfleis** [comment]
Thanks! I can reproduce in the container. That doesn't help me understanding why it happens but it is a step forward.

**Comment by martinfleis** [comment]
I am stupid. 

`GeoSeries.transform` is a new method that is available only on geopandas main branch and will be released as part of GeoPandas 1.0 in about two months. Any older version of geopandas is falling back to `pandas.Series.transform`, which obviously cannot handle geometries.

If you are installing released versions please refer to the stable documentation, which is what geopandas.org shows by default.
